components/layout_2020/flow/inline.rs

```diff
@@ -989,7 +989,8 @@ impl<'a, 'b> InlineFormattingContextState<'a, 'b> {
                 match self.current_line.count_justification_opportunities() {
                     0 => Length::zero(),
                     num_justification_opportunities => {
-                        (available_space - line_length) / (num_justification_opportunities as f32)
+                        (available_space - text_indent - line_length) /
+                            (num_justification_opportunities as f32)
```

Comment:
This was already wrong and can be left for a follow-up, but it should have `.max(Length::zero())`.

```html
<!DOCTYPE html>
<style>div { text-align: justify; border: solid; width: 100px; white-space: pre-wrap; }</style>
<div style="width: 50px">lorem&nbsp;ipsum sit amet</div>
<div style="width: 100px; text-indent: 50px">lorem&nbsp;ipsum sit amet</div>
```
![](https://github.com/servo/servo/assets/7477678/e631e9cb-23ac-45ec-860d-fce5d778e7b9)

Reply:
Thanks for pointing this out. I've added this to the change as well as incorporating your test.